Add woocommerce stats paths

psealock opened the pull request:
### Establish WooCommerce Stats routes
Begin inclusion of [WooCommerce Store Stats](https://github.com/Automattic/wp-calypso/projects/26) into [WooCommerce for Calypso](https://github.com/Automattic/wp-calypso/projects/22). It appears each project will occupy the same space.

### Routes
I've added the beginning of a basic route for `stats` section. I also organised the `/products/add` path to follow the convention set by the rest of Calypso where the `:site` parameter [comes last](https://github.com/Automattic/wp-calypso/blob/master/client/my-sites/stats/index.js).

### Feature Flags
Although the extension is hidden from production builds by the environment id, there may be a scenario where we may want to move forward with one portion before the others are finished. The flags can be changed to reflect that.

Any feedback on naming, conventions, or otherwise are welcome.


cc @justinshreve @coderkevin @greenafrican

coderkevin commented:
Thanks @psealock, I'm happy to work with you here.
@justinshreve has edited these files last, I'd like him to take a look here too.

coderkevin commented on config/development.json:
```diff
@@ -153,6 +153,8 @@
 		"upgrades/presale-chat": true,
 		"webpack/persistent-caching": false,
+		"woocommerce/extension": true,
+		"woocommerce/extension/stats": true,
```
I'm not 100% sure how I feel about this. When I put in the `woocommerce/extension` part of this, it was in lieu of creating a feature flag for each extension, which would mean this line has a 1:1 relationship with the extension itself. If we include another line for a different part of the same extension, I'm not sure how that will work when we actually have feature flags for extensions.

justinshreve commented on client/extensions/woocommerce/app/stats/index.js:
```diff
@@ -0,0 +1,26 @@
+
+	render() {
+		return (
+			<Main className="stats woocommerce__main">
```
Instead of `woocommerce__main`, use `woocommerce` here. We are name-spacing our CSS under it.

justinshreve commented:
> I also organised the /products/add path to follow the convention set by the rest of Calypso where the :site parameter comes last.

I realized this needed to be corrected the other day, so thanks for handling that.

psealock commented:
>I'm not 100% sure how I feel about this. When I put in the woocommerce/extension part of this, it was in lieu of creating a feature flag for each extension, which would mean this line has a 1:1 relationship with the extension itself. If we include another line for a different part of the same extension, I'm not sure how that will work when we actually have feature flags for extensions.

I'm attempting to address the likely scenario that `stats` will be ready to go before the rest of the extension is. In order to avoid implying a 1:1 relationship, I've renamed the flags
```
"woocommerce/extension-dashboard": true,
"woocommerce/extension-stats": true,
```
This way, once the extension is live, `woocommerce/extension-dashboard` can be turned off and the only routes available will be those contained in this if statement
```js
if ( config.isEnabled( 'woocommerce/extension-stats' ) ) {
    page( '/store/stats/:site?', siteSelection, navigation, Controller.stats );
}
```
>Instead of woocommerce__main, use woocommerce here. We are name-spacing our CSS under it.

Cheers, fixed.

coderkevin commented:
This approach seems reasonable. I like the idea of decoupling the releases. Thanks!
LGTM :shipit:
